types/tls.go

```diff
@@ -183,3 +166,26 @@ var (
 		TLS_RSA_WITH_AES_256_CBC_SHA:                  "AES256-SHA",
 	}
 )
+
+func (c *TLSCipherSuite) String() string {
+	return fmt.Sprintf("%s", *c)
+}
+
+// TODO: Should this return an errgroup instead of only the first error?
+func ValidateEnvoyCipherSuites(cipherSuites []TLSCipherSuite) error {
+	for _, c := range cipherSuites {
+		if _, ok := envoyTLSCipherSuiteStrings[c]; !ok {
+			return fmt.Errorf("no matching Envoy TLS cipher suite found for %s", c.String())
+		}
+	}
+
+	return nil
+}
+
+func MarshalEnvoyTLSCipherSuiteStrings(cipherSuites []TLSCipherSuite) []string {
+	cipherSuiteStrings := []string{}
+	for _, c := range cipherSuites {
+		cipherSuiteStrings = append(cipherSuiteStrings, envoyTLSCipherSuiteStrings[c])
```

Comment:
This could potentially append `""` if cipherSuites wasn't validated already.

Reply:
Yeah you want to conditionally append to the list only if there is a valid envoy mapping otherwise skip it.

Elsewhere in the place you call this function you may wish to double check that if you pass it a slice of one cipher suite, and we skip it, that the code is cool getting back an empty slice before xds-ing it.